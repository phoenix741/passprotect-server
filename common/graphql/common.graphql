scalar Date

# Encrypted content
type EncryptedContent {
  # Content encrypted with salt
  content: String!
  # Authentification tag
  authTag: String!
}

# Encrypted content
input EncryptedContentInput {
  # Content encrypted with salt
  content: String!
  # Authentification tag
  authTag: String!
}

type Error {
  fieldName: String,
  message: String
}

type Errors {
  errors: [Error]!
}

schema {
  query: RootQuery
  mutation: RootMutation
}
