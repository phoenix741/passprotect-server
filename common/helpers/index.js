'use strict';

var _ = require('underscore');

module.exports = extendGlobal;

function extendGlobal(globalObject) {
	_.extend(globalObject, {});
}
