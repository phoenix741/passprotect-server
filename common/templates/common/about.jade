.container
	.row
		.col.s12.m9.l10
			h2.main-title.header #{t('about.title')}

			p.caption #{t('about.copyright')}
