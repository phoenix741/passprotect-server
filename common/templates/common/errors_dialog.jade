.modal-content
	h5= title || t('alert.unknown_error.title')

	p!= message

.modal-footer
	button.modal-action.modal-close.waves-effect.waves-green.btn-flat(type='button')= t('alert.unknown_error.close')

