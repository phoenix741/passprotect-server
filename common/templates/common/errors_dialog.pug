.modal-content
	h5.main-title= title || t('alert.unknown_error.title')

	p!= message

.modal-footer
	button.modal-action.modal-close.waves-effect.waves-green.btn-flat(type='button')= t('alert.unknown_error.close')

