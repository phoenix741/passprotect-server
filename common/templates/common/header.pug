.nav-wrapper.container
	a#logo-container.brand-logo(href='#')= t('app.title')

	#navigation-large.right.hide-on-med-and-down
		ul
			li
				a(href='/about')= t('app.menu.about')

	#navigation-mobile.side-nav
		ul
			li.bold
				a(href='/about')= t('app.menu.about')

	a.button-collapse(data-activates='navigation-mobile',href='#')
		i.material-icons menu
