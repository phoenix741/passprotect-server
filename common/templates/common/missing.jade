.content.container
	.alert(role='alert')
		.col.s12
			.card.red.darken-1
				.card-content.white-text
					p= t('page:error.not_found')

