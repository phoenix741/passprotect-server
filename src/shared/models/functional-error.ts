import { HttpException } from '@nestjs/common';

export class FunctionalError extends Error {
  field?: string;

  constructor(field?: string, message?: string) {
    super(message);

    this.field = field;
  }

  toGraphQL(defaultLabel = 'label') {
    return {
      errors: [{
        fieldName: this.field || defaultLabel,
        message: this.message,
      }],
    };
  }
}

export function toFunctionalError(error: Error): FunctionalError {
  if (error instanceof FunctionalError) {
    return error;
  }
  if (error instanceof HttpException) {
    return new FunctionalError(undefined, error.message.message);
  }
  return new FunctionalError(undefined, error.message);
}
