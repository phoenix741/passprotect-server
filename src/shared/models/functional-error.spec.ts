import { toFunctionalError, FunctionalError } from './functional-error';
import { NotImplementedException } from '@nestjs/common';

describe('FunctionalError', () => {
  describe('toFunctionalError', () => {
    it('Create the functional error, from functional error', () => {
      const error = toFunctionalError(new FunctionalError('field', 'mymessage'));
      expect(error).toMatchSnapshot('functional error');
    });

    it('Create the functional error, from http exception', () => {
      const error = toFunctionalError(new NotImplementedException('not implemented'));
      expect(error).toMatchSnapshot('http exception');
    });

    it('Create the functional error, from other exception', () => {
      const error = toFunctionalError(new Error('my error'));
      expect(error).toMatchSnapshot('other exception');
    });
  });

  describe('toGraphQL', () => {
    it('generate error for graphql', () => {
      const error = new FunctionalError('myfield', 'mymessage');
      expect(error.toGraphQL()).toMatchSnapshot('functionnal error graphql');
    });

    it('generate error for graphql 2', () => {
      const error = new FunctionalError(null, 'mymessage');
      expect(error.toGraphQL('my default field')).toMatchSnapshot('functionnal error graphql');
    });
  });
});
