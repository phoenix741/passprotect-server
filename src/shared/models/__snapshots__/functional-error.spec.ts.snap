// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalError toFunctionalError Create the functional error, from functional error: functional error 1`] = `[Error: mymessage]`;

exports[`FunctionalError toFunctionalError Create the functional error, from http exception: http exception 1`] = `[Error: not implemented]`;

exports[`FunctionalError toFunctionalError Create the functional error, from other exception: other exception 1`] = `[Error: my error]`;

exports[`FunctionalError toGraphQL generate error for graphql 2: functionnal error graphql 1`] = `
Object {
  "errors": Array [
    Object {
      "fieldName": "my default field",
      "message": "mymessage",
    },
  ],
}
`;

exports[`FunctionalError toGraphQL generate error for graphql: functionnal error graphql 1`] = `
Object {
  "errors": Array [
    Object {
      "fieldName": "myfield",
      "message": "mymessage",
    },
  ],
}
`;
